components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -147,7 +149,8 @@ internal object TabListReducer {
 
                     state.copy(
                         tabs = updatedTabList,
-                        selectedTabId = updatedSelection
+                        selectedTabId = updatedSelection,
+                        tabPartitions = state.tabPartitions.removeTabs(action.tabIds)
```

Comment:
When using "undo" to get the tab back, it will not automatically be back in its group. But I guess that is totally fine, since a grouping middleware could observe `TabListAction.RestoreAction` and group the tab again. For groups without a middleware (user managed groups?) this may not work though. However this is also not needed. So maybe just something to file/think about?

Reply:
Yes, I also realized last night that the middlewares will need to react to `RestoreAction` and put tabs back in their original groups. Luckily that's relatively easy for the groups we have now. If the groups are not persisted the middlewares will also need to observe `RestoreCompleteAction` to build up the groups again after startup so at least that's consistent, but yes, would be nice if this could happen automatically later.